packages/shared/src/hooks/integrations/useIntegrationQuery.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from '@tanstack/react-query';
+import { UserIntegration } from '../../graphql/integrations';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { gqlClient } from '../../graphql/common';
+import { USER_INTEGRATION_BY_ID } from '../../graphql/users';
+import { useAuthContext } from '../../contexts/AuthContext';
+
+export type UseIntegrationQueryProps = {
+  id?: string;
+  queryOptions?: UseQueryOptions<UserIntegration>;
+};
+
+export type UseIntegrationQuery = UseQueryResult<UserIntegration>;
+
+export const useIntegrationQuery = ({
+  id,
+  queryOptions,
+}: UseIntegrationQueryProps = {}): UseIntegrationQuery => {
+  const { user } = useAuthContext();
+  const enabled = !!user && !!id;
+
+  const queryResult = useQuery(
+    generateQueryKey(RequestKey.UserIntegrations, user, id),
```

Comment:
Shouldn't the query params be part of the key? 

```suggestion
    generateQueryKey(RequestKey.UserIntegrations, user, id, queryOptions),
```

Reply:
Think it's fine for this specific one.
CC: @capJavert unless you think different?

Reply:
Only things used inside the handler should be used inside query key

Reply:
Ah, yes, my bad, thought they were params of the actual query. Lgtm! 🚀